test/e2e/performanceprofile/functests/1_performance/cpu_management.go

```diff
@@ -615,6 +618,86 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", Ordered, func() {
+			// and runtime class. This is required as the cpu id's used by the container are
+			// written to cpuset.cpus.exclusive
+			const DeploymentName = "test-deployment"
+			var numberofReplicas int32 = 2
+			var dp *appsv1.Deployment
+			annotations := map[string]string{
+				"cpu-load-balancing.crio.io": "disable",
+			}
+			p := pods.GetTestPod()
+			p.Spec.NodeSelector = testutils.NodeSelectorLabels
+			runtimeClass := components.GetComponentName(profile.Name, components.ComponentNamePrefix)
+			p.Spec.RuntimeClassName = &runtimeClass
+			p.Spec.Containers[0].Image = images.Test()
+			p.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceMemory: resource.MustParse("200Mi"),
+					corev1.ResourceCPU:    resource.MustParse("2"),
+				},
+			}
+
+			// we create and delete deployment in loop to create deployments in
+			// quick succession to verify pre-start hook is able to write to
+			// cpuset.cpus.exclusive
+			for i := 0; i < 5; i++ {
+				testlog.Infof("%d Create deployment %s with 2 Guaranteed pods requesting 2 cpus", i, DeploymentName)
+				dp = deployments.Make(DeploymentName, testutils.NamespaceTesting,
+					deployments.WithPodTemplate(p),
+					deployments.WithNodeSelector(testutils.NodeSelectorLabels))
+				dp.Spec.Template.Annotations = annotations
+				dp.Spec.Replicas = &numberofReplicas
+				Expect(testclient.DataPlaneClient.Create(ctx, dp)).ToNot(HaveOccurred())
+				podList := &corev1.PodList{}
+				listOptions := &client.ListOptions{Namespace: testutils.NamespaceTesting, LabelSelector: labels.SelectorFromSet(dp.Spec.Selector.MatchLabels)}
+				Eventually(func() bool {
+					if err := testclient.DataPlaneClient.List(context.TODO(), podList, listOptions); err != nil {
+						return false
+					}
+					for _, pod := range podList.Items {
+						Expect(pod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+						if pod.Status.Phase != corev1.PodRunning && !isPodReady(&pod) {
+							for _, containerStatus := range pod.Status.ContainerStatuses {
+								if containerStatus.State.Waiting != nil {
+									if containerStatus.State.Waiting.Reason == "RunContainerError" && strings.Contains(containerStatus.State.Waiting.Message, "failed to run pre-start hook for container") {
+										testlog.Infof("container %s failed to start with error: %s", pod.Spec.Containers[0].Name, containerStatus.State.Waiting.Message)
+										return false
+									}
+								}
+							}
+						}
+					}
+					if len(podList.Items) < 1 {
+						return false
+					}
+					for _, s := range podList.Items[0].Status.ContainerStatuses {
+						if s.Ready == false {
+							return false
+						}
+					}
+					return true
+				}, 10*time.Second, 5*time.Second).Should(BeTrue())
+				// delete deployment
+				time.Sleep(5 * time.Second)
+				testlog.Infof("Deleting Deployment %v", dp.Name)
+				err := testclient.Client.Delete(ctx, dp)
```

Comment:
Since right after the deletion you're about to create the `Deployment` again, I would check that it's actually gone completely, by trying to `Get` it and expect `IsNotFound` error or at least move the cooling time from line 694 to here.

Reply:
okay we are creating a deployment in quick succession because that's what is required to reproduce the issue.  I updated the PR description but let me put it again. 

this PR automates the bug OCPBUGS-34812 where we need to create  and delete deployment in quick succession so that the old cgroup of the delete pod which still has access to cpus is not released causing the new cgroup creation fail as the pre start hook fails to update cpuset.cpus.exclusive file.